### ISSUE TITLE
score_tests are all negative

### ISSUE THREAD

**Issue by FeiyangBai** [autogluon/autogluon#3808]
Hi AutoGluon team, 

Thanks for making this AutoGluon for us to use. I heard AutoGluon at a conference in New Orleans this week. 

However, I ran some of my data with the code in Colab. It ran well, however, the score_test values are all negative (see below link) 
https://colab.research.google.com/drive/1QznEsKdz8MyymerQ7W76kWJeiybjSjie#scrollTo=AfJX-sv7ZJY2 

I don't know what I did wrong. It would be very helpful if you could help me diagnose this problem. 

Thanks, 

Feiyang

**Comment by shchur** [comment]
Hi Feiyang, this is by design – AutoGluon always reports scores in higher-is-better format. Therefore, negatively oriented scores like RMSE are multiplied by -1 to follow the higher-is-better format. This means in case of RMSE the `score_test` column will actually contain "negative RMSE" and not RMSE.

**Comment by FeiyangBai** [comment]
Hi Shchur, 

Thanks for your prompt response. Other couples more questions please: 

1. https://auto.gluon.ai/stable/tutorials/tabular/tabular-quick-start.html; In this AutoGluon Tabular tutorial, everything is the same with my code, except I change the data set to my own. So if this is by design, why the tutorial's values are all positive?
2.  https://auto.gluon.ai/stable/tutorials/tabular/tabular-quick-start.html; in the tutorial, this line of code #predictor.evaluate(test_data, silent=True)# generate following: 
{'accuracy': 0.949,
 'balanced_accuracy': 0.7568673702569989,
 'mcc': 0.9374979113792482}
while mine generate: 
{'root_mean_squared_error': -36.79356437040658,
 'mean_squared_error': -1353.7663790792526,
 'mean_absolute_error': -35.35441780938095,
 'r2': -5.529278404794967,
 'pearsonr': 0.7253843100989322,
 'median_absolute_error': -34.950264553494314}

why did this happen? And especially since all my values are all negative. 

3. If this is my data set's problem, I ran the same experiment with the Matlab regression learner app. I got around 96 of the accuracy. Although the algorithms here might be different, even if I multiply the results by -1; the results here are still hard to watch compared to the results I got from Matlab. 

Thank you so much for your kindness in answering my questions. 

Feiyang
5.
6.

**Comment by shchur** [comment]
In your dataset, you are solving a regression problem (predicting a continuous value). The tutorial considers a classification problem (predicting a categorical variable). The metrics for classification like accuracy are in higher-is-better format by default, so they don't need to be multiplied by -1. In contrast, most regression metrics are in lower-is-better format by default, so they need to be multiplied by -1 to convert to the higher-is-better format.

**Comment by FeiyangBai** [comment]
Hi Shchur, 

Thanks for the prompt response. I am in a conference during the day, so I respond slowly.  But 2 more questions please: 

1. I read the description again here (https://auto.gluon.ai/stable/tutorials/tabular/tabular-quick-start.html). The example is also a regression problem. Please see the first paragraph I copied from the AutoGluon Tabular - Quick Start. 
"In this tutorial, we will see how to use AutoGluon’s TabularPredictor to predict the values of a target column based on the other columns in a tabular dataset."

2. Let's say that I need to multiply -1 to my result. But my best result is NeuralNetTorch (72.808196%). As I mentioned, I tested the data in Matlab with the best behavior with the GPR algorithm at 96%. I think this is a little hard to believe. The difference between different algorithms will be that big. 

Sorry for keep bothering you. But thank you for your patience in answering my questions. 

Thank you. 

Feiyang

**Comment by shchur** [comment]
1. The column has categorical values, so it's a classification problem. In your example the values are continuous, so it's regression.
2. Make sure that you are comparing scores computed on an unseen test set. Otherwise the train score can be much better than the validation score.

**Comment by FeiyangBai** [comment]
Thank you so much.

1. How can I change the code from classification to regression?
2. Yeah, the test data are unseen in the training dataset.

Thanks.

Feiyang
